vendor/bat-native-ledger/src/bat/ledger/internal/state/state.cc

```diff
@@ -83,6 +84,53 @@ base::flat_map<std::string, std::string> StringToPayoutStatus(
   return payout_status;
 }
 
+std::string WalletProviderRegionsToString(
+    const base::flat_map<std::string, ledger::mojom::RegionsPtr>&
+        wallet_provider_regions) {
+  base::Value::Dict wallet_provider_regions_dict;
+
+  for (const auto& [wallet_provider, regions] : wallet_provider_regions) {
+    base::Value::List allow;
+    for (const auto& country : regions->allow) {
+      allow.Append(country);
+    }
+
+    base::Value::List block;
+    for (const auto& country : regions->block) {
+      block.Append(country);
+    }
+
+    base::Value::Dict regions_dict;
+    regions_dict.Set("allow", std::move(allow));
+    regions_dict.Set("block", std::move(block));
+
+    wallet_provider_regions_dict.Set(wallet_provider, std::move(regions_dict));
+  }
+
+  std::string string;
+  base::JSONWriter::Write(wallet_provider_regions_dict, &string);
```

Comment:
I'm not entirely happy with our existing practice of storing stringified JSON in the `Preferences` file. It's already JSON, so intuitively we should be able to store structured data there. Is there any way that we can avoid storing this as stringified JSON?

Reply:
What currently prevents us from doing that is that we don't have a
```
[Sync]
SetValueState(string name, mojo_base.mojom.Value value) => ();
```
request on the `BatLedgerClient` interface in `bat_ledger.mojom`, with which we could send `base::Value::List`s, or `base::Value::Dict`s from the utility process to the browser process, but I can take care of that.